### PR TITLE
Use [all] to bring in all dependencies, and rename old [all] to [extra]

### PR DESCRIPTION
**Pull request by nvaytet**
The fact that `pip install scipp[all]` does not bring in all optional dependencies is somewhat confusing.
Having to do `pip install scipp[all, interactive]` is somewhat cumbersome.

With this PR, `pip install scipp[all]` now pulls in all optional deps (including libs for interactive plotting), and we now use `[extra]` instead of the old `[all]`.

### CODE REVIEW

File: docs/getting-started/installation.rst [hunk 1]
```diff
@@ -78,19 +78,19 @@ Scipp is available from `PyPI <https://pypi.org/>`_ via ``pip``:
...
 .. code-block:: sh
 
    pip install scipp[all]
 
-This will install everything that is needed, with the exception of plotting libraries for interactive use in Jupyter.
-These are available in the ``interactive`` extra.
-We recommend combining this with ``all``:
+
+You can also leave out the interactive tools and bring in only functional optional dependencies,
+such as ``h5py`` and ``scipy``, use ``extra`` instead of ``all``:
```
Comment (jl-wynen) [inline docs/getting-started/installation.rst:89]: It would still be useful to mention `interactive` as a possibility.

Comment (nvaytet) [reply docs/getting-started/installation.rst:89]: Is it really worth it? Would you really install `interactive` without `extra`? I thought if you've decided to install all the interactive stuff, you probably don't mind the few extra packages in `extra`?

Leaving it out has the advantage of making this page a little less long/confusing.

Comment (jl-wynen) [reply docs/getting-started/installation.rst:89]: But then we should remove the `interactive` extra altogether?

Comment (nvaytet) [reply docs/getting-started/installation.rst:89]: We could. What would be the syntax in `pyproject` then? Not sure I know how...

Comment (jl-wynen) [reply docs/getting-started/installation.rst:89]: I would think it is just
```toml
all = [
  "scipp[extra]",
  "h5py",
  ...
]
```

**Review by jl-wynen — APPROVED** [review]

**Comment by nvaytet** [comment]
I tested this locally and it does seem to fetch all dependencies including plotting stuff when doing `pip install scipp[all]`.

**Pull request merged**